### PR TITLE
Data design deep dive overhaul

### PR DESCRIPTION
**Pull request by thoragudf**
New videos, updated steps according to changes in the produt. Please check if I forgot something!

### CODE REVIEW

**Comment by hugihlynsson** [comment]
#### 📝 Changed routes:


| [**…-tracking-plan/metrics**](https://docs-eg8irrp2p.avo-dev.app/data-design/avo-tracking-plan/metrics) | [**…sign/start-data-design**](https://docs-eg8irrp2p.avo-dev.app/data-design/start-data-design) |
|:-:|:-:|
|<a href="https://docs-eg8irrp2p.avo-dev.app/data-design/avo-tracking-plan/metrics">Screenshot of /data-design/avo-tracking-plan/metrics</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/_Yy9Mj1hr7jE_Be5VlzETf2FRRmHJXEdruFTkKyjidgPGhJzhD37LO3YG573KXAvxCRWUUdfLyhWoycgTGhQDw.png">(view full size)</a> |<a href="https://docs-eg8irrp2p.avo-dev.app/data-design/start-data-design">Screenshot of /data-design/start-data-design</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/WIdasJ56Ja3c7I_gVD5f-Q6Zuso60Q0XC1jup3cB_Tpv26J-smlZtgPWH5AQQAEfmjgjOvh3CDC9PGU6wteInw.png">(view full size)</a> |







Commit 307033c021e9901238ee63b01b09e23865cba6c3 (https://docs-eg8irrp2p.avo-dev.app).

File: pages/data-design/start-data-design.mdx [hunk 1]
```diff
@@ -13,98 +15,95 @@
...
+We always recommend starting with the why in data design. That means thinking about your research questions, goals and metrics before your start defining your event structures. Metrics are directly tied to events in Avo and are visible in the event details. Read more about metrics, why they are useful and learn about purpose meetings in our [Metrics page](/data-design/avo-tracking-plan/metrics).
 
 ### Step 1 – Open a branch in Avo
 
 If you are making changes to your tracking plan in Avo, we always recommend creating a branch first – no matter how small your change might be. That allows you to make suggestions in isolation from the source of truth on your main branch and get feedback from your peers.
 
-Open a new branch by clicking the three line icon beside where it says "main" in the left sidebar. Then you are prompted to open an existing branch or creating a new one.
+Open a new branch by clicking the icon "main" in the left sidebar. Then you are prompted to open an existing branch or creating a new one.
```
Comment (hugihlynsson) [inline pages/data-design/start-data-design.mdx:24]:
```suggestion
Open a new branch by clicking the icon to the right of "main" in the left sidebar. Then you are prompted to open an existing branch or creating a new one.
```

File: pages/data-design/start-data-design.mdx [hunk 2]
```diff
@@ -307,13 +375,52 @@ To add an existing event property to your event, click "+ Add Event Property" un
...
+    image={'/docs/images/svg/analyticsmanager.svg'}
+    title="Requesting implementation"
+    description="How to request implementation in Avo."
+    href={'/workflow/request-implementation'}
+  />
+  <PageLink
+    image={'/docs/images/svg/analyticsmanager.svg'}
+    title="Sources and destinations"
+    description="How to set up your sources and destinations in Avo."
+    href={'data-design/define-sources-and-destinations'}
```
Comment (hugihlynsson) [inline pages/data-design/start-data-design.mdx:423]: 404, probably needs a leading slash:
```suggestion
    href="/data-design/define-sources-and-destinations"
```

**Review by hugihlynsson — COMMENTED** [review]
Looking good, the new videos are 🔥🔥🔥

File: pages/data-design/avo-tracking-plan/metrics.mdx [hunk 3]
```diff
@@ -1,12 +1,50 @@
+import PageLink, { TwoCol } from '../../../components/PageLink';
+
 # Metrics
 
 *Learn about the metrics in your tracking plan*
 
+We always recommend starting with the why in data design. That means thinking about your research questions, goals and metrics before your start defining your event structures. That way you are more likely to **design the events and properties you need to measure the success of your product update** – and won't end up with designing and implementing data structures that you don't have any use for. The added benefit of documenting metrics in context of your events is **knowing what metrics are affected by changing this event**. Read more in our [documenting downstream dependencies guide](/data-design/documenting-downstream-dependencies).
+
+At Avo, we have developed a framework that we call the **Purpose Meeting** to help us align on goals and metrics. Read more in the blog post about [tracking the right product metrics]() by our CEO, Stef.
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/data-design/avo-tracking-plan/metrics.mdx:9]: eagle eyes

File: pages/data-design/avo-tracking-plan/metrics.mdx [hunk 4]
```diff
@@ -1,12 +1,50 @@
...
+    image={'/docs/images/svg/analyticsmanager.svg'}
+    title="Organizing Metrics and Events"
+    description="How to organize metrics and events in Avo."
+    href={'/data-design/organizing-metrics-and-events'}
+  />
+  <PageLink
+    image={'/docs/images/svg/analyticsmanager.svg'}
+    title="Documenting Purpose Meetings in Avo"
+    description="How to document the result of a purpose meeting in Avo."
+    href={'/docs/data-design/guides/documenting-purpose-meetings-in-avo'}
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/data-design/avo-tracking-plan/metrics.mdx:48]: correct!

**Review by hugihlynsson — APPROVED** [review]

File: pages/data-design/start-data-design.mdx [hunk 5]
```diff
@@ -240,60 +298,70 @@ In this guide we're focusing on
...
-Change the data type of your property to match the one you need by clicking on where it says "String" below the description input. Then you will be able to pick from the following types:
+Change the [data type of your property](/data-design/avo-tracking-plan/properties#property-types-and-constraints) to match the one you need by clicking on where it says "String" below the description input. Then you will be able to pick from the following types:
+
+- **string**: only all string values are allowed
+- **int**: only whole number values are allowed
+- **float**: only floating point values are allowed
+- **boolean**: only `true` or `false` values are allowed
+- **[object](/data-design/object-properties)**: only dictionary or nested object structures are allowed (learn more in our
+  [object docs](/data-design/object-properties))
+- **any** (available upon request): any data type is allowed and not type validation will occur
```
Comment (1212spark) [inline pages/data-design/start-data-design.mdx:331]: "not" --> "no"

File: pages/data-design/start-data-design.mdx [hunk 6]
```diff
@@ -240,60 +298,70 @@ In this guide we're focusing on
...
+The [property presence](/data-design/avo-tracking-plan/properties#configuring-when-properties-are-required-or-optional) defines whether the property is required, optional or not allowed. In Avo it is described with `Always sent` which means the property is required, `Sometimes sent` which means it's optional and `Never sent` which means is not allowed. Presence can be configured per event and per source.
+
+All properties are required by default and we recommend trying to keep them required in as many cases a possible. That means that the Avo developer tools will throw and error if the property is not assigned a value in development, which reduces the risk of missing data significantly. It the property is only applicable in some cases, we recommend using [event variants](/data-design/avo-tracking-plan/event-variants) to define each case. It's also possible to use `Sometimes sent`, but that means that the validation will not be as rigorous.
```
Comment (1212spark) [inline pages/data-design/start-data-design.mdx:339]: "cases a possible" --> "cases as possible"

Comment (1212spark) [reply pages/data-design/start-data-design.mdx:339]: "throw and error" --> "throw an error"

File: pages/data-design/start-data-design.mdx [hunk 7]
```diff
@@ -307,13 +375,52 @@ To add an existing event property to your event, click "+ Add Event Property" un
...
+  alt="Adding existing properties to an event"
+/>
+
+_Example: Creating and adding a property bundle to an event._
+
 ## What's next?
 
-Now you know how to define and maintain your metrics and events in Avo and it's time to learn
-[how to manage sources and destinations](/data-design/define-sources-and-destinations).
+Now you know how to define and maintain your metrics and events in Avo and it's time to learn how to [review branch changes](/workflow/review) and [request implementation](/workflow/request-implementation). If you haven't set up your sources and destinations yet, now is a good time to learn how [sources and destinations](data-design/define-sources-and-destinations) are defined.
```
Comment (1212spark) [inline pages/data-design/start-data-design.mdx:404]: "Now **that** you know . . . in Avo, it's time"

**Review by 1212spark — CHANGES_REQUESTED** [review]
Some of these are stylistic changes and some are typos; good to go after review and accept/reject

File: pages/data-design/avo-tracking-plan/metrics.mdx [hunk 8]
```diff
@@ -1,12 +1,50 @@
...
+
+At Avo, we have developed a framework that we call the **Purpose Meeting** to help us align on goals and metrics. Read more in the blog post about [tracking the right product metrics](https://www.avo.app/blog/tracking-the-right-product-metrics) by our CEO, Stef.
+
+Metrics as well as purpose meetings can be documented in Avo.
+
+## What are metrics in Avo?
+
 Metrics are used to measure the success of a feature. Usually more than one metrics are necessary to be able to evaluate the success of a feature. Example metrics for a feature that is meant to improve signup experience are:
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/data-design/avo-tracking-plan/metrics.mdx:15]: Changed to: "How do metrics in Avo work?"

File: pages/data-design/start-data-design.mdx [hunk 9]
```diff
@@ -240,60 +298,70 @@ In this guide we're focusing on
...
 
-As with event, we highly recommend adding descriptions to all properties. That helps the ones that implement to know exactly what variable in their code they should assign to that property.
+As with events, we highly recommend adding descriptions to all properties. That creates a common understanding between the data designer and those that use the data or implement it. It's especially important to make sure the implementation is consistent across products, platforms, events and code paths as the description helps the ones that implement to know exactly what variable in their code they should assign to that property.
 
 #### Select the data type
 
-Change the data type of your property to match the one you need by clicking on where it says "String" below the description input. Then you will be able to pick from the following types:
+Change the [data type of your property](/data-design/avo-tracking-plan/properties#property-types-and-constraints) to match the one you need by clicking on where it says "String" below the description input. Then you will be able to pick from the following types:
+
+- **string**: only all string values are allowed
```
[the remark this answers is not in the document]
Comment (thoragudf) [reply pages/data-design/start-data-design.mdx:325]: I think "only string values are allowed" does the trick well enough

**Review by 1212spark — APPROVED** [review]
🚀

**Pull request merged**